src/bandersnatch/tests/plugins/test_metadata_plugins.py

```diff
@@ -0,0 +1,130 @@
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import cast
+from unittest import TestCase
+
+from mock_config import mock_config
+
+import bandersnatch.filter
+from bandersnatch.master import Master
+from bandersnatch.mirror import BandersnatchMirror
+from bandersnatch.package import Package
+from bandersnatch_filter_plugins.metadata_filter import SizeProjectMetadataFilter
+
+
+class TestSizeProjectMetadataFilter(TestCase):
+    """
+    Tests for the bandersnatch filtering by project size
+    """
+
+    def setUp(self) -> None:
+        self.cwd = os.getcwd()
+        self.tempdir = TemporaryDirectory()
+        os.chdir(self.tempdir.name)
+
+    def tearDown(self) -> None:
+        if self.tempdir:
+            assert self.cwd
+            os.chdir(self.cwd)
+            self.tempdir.cleanup()
```

Comment:
Do your tests actually write out any files? I never understood if these are actually needed. I'm guessing you just copy pasted from other test, but was just wondering. I guess it will write out some base mirror directory structure.

Reply:
it's true. I'm guilty of copy/pasting. I just presumed that it was a required practice I should follow. :)

Reply:
And [to some](url), not best practice-y enough since `tearDown()` won't be called if `setUp()` fails.

Reply:
pytest is lots of voodoo. Maybe we can make an issue to test / clean these up.